iaf-aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/services/conv/impl/convertors/AbstractConvertor.java

```diff
@@ -0,0 +1,179 @@
+	 * updated (metaData set and any attachments added).
+	 * 
+	 * @param mediaType
+	 * @param conversionOption
+	 */
+	protected abstract void convert(MediaType mediaType, File file, CisConversionResult builder,
+			ConversionOption conversionOption) throws Exception;
+
+	@Override
+	public List<MediaType> getSupportedMediaTypes() {
+		return supportedMediaTypes;
+	}
+
+	private void checkForSupportedMediaType(MediaType mediaType) {
+
+		boolean supported = false;
+
+		for (MediaType mediaTypeSupported : getSupportedMediaTypes()) {
+			if (mediaTypeSupported.equals(mediaType)) {
+				supported = true;
+				break;
+			}
+		}
+
+		// Create informative message.
+		if (!supported) {
+			StringBuilder builder = new StringBuilder();
+			builder.append("This convertor '");
+			builder.append(this.getClass().getSimpleName());
+			builder.append("' only supports ");
+
+			boolean first = true;
+			for (MediaType mediaTypeSupported : getSupportedMediaTypes()) {
+				if (!first) {
+					builder.append(" or");
+				}
+				first = false;
+				builder.append(" '");
+				builder.append(mediaTypeSupported.toString());
+				builder.append("'");
+			}
+			builder.append("! (received '");
+			builder.append(mediaType);
+			builder.append("')");
+			throw new IllegalArgumentException(builder.toString());
+		}
+
+	}
+
+	/**
+	 * Should not be overloaded by the concrete classes.
+	 */
+	@Override
+	public final CisConversionResult convertToPdf(MediaType mediaType, String filename, File file,
+			ConversionOption conversionOption) {
+
+		checkForSupportedMediaType(mediaType);
+
+		CisConversionResult result = new CisConversionResult();
+		File resultFile = null;
+		try {
+			resultFile = UniqueFileGenerator.getUniqueFile(pdfOutputlocation, this.getClass().getSimpleName(), "pdf");
+			result.setConversionOption(conversionOption);
+			result.setMediaType(mediaType);
+			result.setDocumentName(ConvertorUtil.createTidyNameWithoutExtension(filename));
+			result.setPdfResultFile(resultFile);
+			result.setResultFilePath(resultFile.getAbsolutePath());
+			
+			LOGGER.debug("Convert to file... " + file.getName());
+			convert(mediaType, file, result, conversionOption);
+			LOGGER.debug("Convert to file finished. " + file.getName());
+
+		} catch (Exception e) {
+			if (isPasswordException(e)) {
+				result = CisConversionResult.createPasswordFailureResult(filename, conversionOption, mediaType);
+			} else {
+				result.setFailureReason(createTechnishefoutMsg(e));
+			}
+			// Clear the file to state that the conversion has failed.
+			result.setPdfResultFile(null);
+		}
+		return result;
+	}
+
+	protected String getPdfOutputlocation() {
+		return pdfOutputlocation;
+	}
+	
+	protected int getNumberOfPages(File file) {
+		int result = 0;
+		if(file != null) {
+			try (InputStream inStream = new FileInputStream(file)){
+				Document doc = new Document(inStream);
+				result = doc.getPages().size();
+				System.out.println(" get number of pages has ben executed  "+result);
```

Comment:
Code should not contain System.out.println(). Please remove all occurrences, or replace with appropriate logging.